packages/smooth_app/lib/pages/scan/mkit_scan_helper.dart

```diff
@@ -233,13 +233,21 @@ class _MLKitScanDecoderIsolate {
 
     port.send(
       barcodes
-          .map((Barcode barcode) => barcode.rawValue)
+          .map((Barcode barcode) => _changeBarcodeType(barcode))
           .where((String? barcode) => barcode?.isNotEmpty == true)
           .cast<String>()
           .toList(growable: false),
     );
   }
 
+  static String? _changeBarcodeType(Barcode barcode) {
+    //EAN13 begins with 0 is detected as UPC-A by google_mlkit_barcode_scanning v0.3.0
+    if (barcode.format == BarcodeFormat.upca && barcode.rawValue != null) {
```

Comment:
Sorry for the second review, but you should invert the conditions.
If the first one (!= null) is false, it won't compute the second

Reply:
not the same attribute 😄 , It might not trigger the short circuit ?

Reply:
I know, but i rawValue == null, then the second one won't be computed
And that's what we want

Reply:
✅